srcts/src/components/card.ts

```diff
@@ -184,6 +187,25 @@ class Card {
     document.body.classList.remove(Card.attr.CLASS_HAS_FULL_SCREEN);
 
     this._emitFullScreenEvent(false);
+    this._setShinyInput();
+  }
+
+  private _setShinyInput(): void {
+    if (!this.card.classList.contains(Card.attr.CLASS_SHINY_INPUT)) return;
+    if (!Shiny) return;
+    if (!Shiny.setInputValue) {
+      if (Shiny.shinyapp) {
+        Shiny.shinyapp.taskQueue.enqueue(() => this._setShinyInput());
+      } else {
+        setTimeout(() => this._setShinyInput(), 0);
+      }
+      return;
+    }
+    Shiny.setInputValue(this.card.id, {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      full_screen:
+        this.card.getAttribute(Card.attr.ATTR_FULL_SCREEN) === "true",
+    });
```

Comment:
Pull this logic out into a `_doSetShinyInput` method. That way `_setShinyInput` doesn't need to be recursive

Reply:
This is intentional and I want `_setShinyInput()` to be recursive. If we have `Shiny` but it doesn't have the `setShinyInput()` method yet, we don't have a reliable non-jQuery way to wait until `Shiny` is fully ready. Neither approach is guaranteed to cause `_setShinyInput()` to run when `Shiny` is fully ready and they may need another cycle or two before we get there.

Happy to add a comment about this in the source.